src/misc/roctx.cc

```diff
@@ -96,4 +94,42 @@ void stringify(roctxPayloadInfo_t payloadInfo) {
   }
 
   snprintf(payloadInfo->message + offset, MAX_MESSAGE_LENGTH - offset, "}");
+}
+
+RCCL_PARAM(LogRoctx, "LOG_ROCTX", 0);
+
+roctx_scoped_range_in::roctx_scoped_range_in(const nvtxPayloadSchemaEntry_t* schema, const nvtxPayloadData_t* data, 
+                                const size_t numEntries, const char* schemaName) noexcept {
+  if (rcclParamLogRoctx()) {
+    roctxAlloc(&payloadInfo, numEntries);
+    extractPayloadInfo(schema, data, numEntries, schemaName, &payloadInfo);
+#ifdef ROCTX_ENABLE
+    roctxRangePushA(payloadInfo.message);
+#else
+    WARN("ROCTX_ENABLE is not defined. Please rebuild with -DROCTX_ENABLE=ON");
+#endif
+  }
+}
+
+roctx_scoped_range_in::roctx_scoped_range_in(const char* message) noexcept {
+  if (rcclParamLogRoctx()) {
+#ifdef ROCTX_ENABLE
+    roctxRangePushA(message);
+#else
+    WARN("ROCTX_ENABLE is not defined. Please rebuild with -DROCTX_ENABLE=ON");
+#endif
+  }
+}
+
+roctx_scoped_range_in::roctx_scoped_range_in() noexcept : roctx_scoped_range_in{""} {/*no impl*/}
+
+roctx_scoped_range_in::~roctx_scoped_range_in() noexcept {
+  if (rcclParamLogRoctx()) {
+#ifdef ROCTX_ENABLE
+    roctxRangePop();
+#else
+    WARN("ROCTX_ENABLE is not defined. Please rebuild with -DROCTX_ENABLE=ON");
+#endif
+    roctxFree(&payloadInfo);
```

Comment:
Should the free not happen within the ifdef?
```suggestion
#ifdef ROCTX_ENABLE
    roctxRangePop();
    roctxFree(&payloadInfo);
#else
    WARN("ROCTX_ENABLE is not defined. Please rebuild with -DROCTX_ENABLE=ON");
#endif
```

Reply:
Depends on the above comment :)